elki-docutil/src/main/java/de/lmu/ifi/dbs/elki/application/internal/DocumentReferences.java

```diff
@@ -333,7 +326,7 @@ public int compare(Pair<Reference, TreeSet<Object>> p1, Pair<Reference, TreeSet<
        * @return Order
        */
       private int compareNull(String s1, String s2) {
-        return (s1 == s2) ? 0 //
```

Comment:
NO. The comparison happens below, the identity is only an optimization for the common case of `s1==s2`.